dev/com.ibm.ws.security.javaeesec.cdi/src/com/ibm/ws/security/javaeesec/cdi/extensions/JavaEESecCDIExtension.java

```diff
@@ -333,7 +335,25 @@ private void createModulePropertiesProviderBeanForApplicationAuthMechToAdd(BeanM
                 e.printStackTrace();
             }
         }
-        addAuthMech(applicationName, implClass, implClass, props);
+
+        Class<?> existingAuthMechClass = httpAuthenticationMechanismsTracker.getExistingAuthMechClass(applicationName);
+        boolean authMechAlreadyExists = (existingAuthMechClass != null);
+        if (authMechAlreadyExists) {
+            boolean isAlternative = (alternative != null);
+            if (isAlternative) {
```

Comment:
An enhancement is needed. Both blocks, alternative or not, could work off each other. If alternative, check if existing mechanism of super class to remove. If not alternative, check if there was an alternative for this class before adding. This would require tracking the class the alternative extends, perhaps in the ModuleProperties.